src/log-query/src/log_query.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chrono::{DateTime, Datelike, Duration, NaiveDate, NaiveTime, TimeZone, Utc};
+use table::table_name::TableName;
+
+use crate::error::{
+    EndBeforeStartSnafu, InvalidDateFormatSnafu, InvalidSpanFormatSnafu, InvalidTimeFilterSnafu,
+    Result,
+};
+
+/// GreptimeDB's log query request.
+pub struct LogQuery {
+    /// A fully qualified table name to query logs from.
+    pub table_name: TableName,
+    /// Specifies the time range for the log query. See [`TimeFilter`] for more details.
+    pub time_filter: TimeFilter,
+    /// Columns with filters to query.
+    pub columns: Vec<ColumnFilters>,
+    /// Maximum number of logs to return. If not provided, it will return all matched logs.
+    pub limit: Option<usize>,
```

Comment:
Maybe we should set the default to 1000 lines or so. It's easy to write a careless query and result in huge returning lines during debugging. Also, it seems a common way to do it (with default limits).
Explicitly set to `0` to return all lines.

Reply:
Good suggestion 👍 The default behavior is not going to be placed in the query definition. I'll set one on executing